### PR TITLE
use host_cpu in BUILD.gn to determine where to find gen_snapshot

### PR DESCRIPTION
**Pull request by flar**
Fixes: https://github.com/flutter/flutter/issues/108238

### CODE REVIEW

File: sky/tools/create_macos_gen_snapshots.py
Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py:25]: I'll wait to chat about this on Monday so I'm sure I have what you are saying right before proceeding.

Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py:25]: The result of get_label_info was `//out/android_debug_unopt/clang_arm64/obj/third_party/dart/runtime/bin` so we'd have to dig the clang_arm64 out of that.

Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py:25]: If I rebase that then I get clang_arm64 as the first element of the path.

Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py:25]:
```
platform: //out/android_debug_unopt/clang_arm64/obj/third_party/dart/runtime/bin
platform rebased: clang_arm64/obj/third_party/dart/runtime/bin
```

Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py:25]: I think I have a solution - just check "host_cpu" in the Build.gn file and pass in either "clang_x64" or "clang_arm"...?

Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py]: I pushed a new solution based on "host_cpu" in the BUILD.gn file.

Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py]: Once more with actual changes this time... (not sure what happened with the last push).

File: sky/tools/create_macos_gen_snapshots.py
Comment (zanderso) [inline sky/tools/create_macos_gen_snapshots.py:19]: The bots call this script, so to make this a soft transition, the new flag would need to be option with a default value of 'clang_x64'.

**Review by zanderso — APPROVED** [review]
lgtm after fix noted below. I'll follow-up with changes to the GN to compute the value for the flag instead of hardcoding it.

File: sky/tools/create_macos_gen_snapshots.py
Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py:19]: Yay! That explains the mystery errors that had no logs.

Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py:19]: But this begs the question - if/when we start having Apple Silicon machines running our CI, won't this just shift the problem from needing a default to needing the create_macos_gen_snapshots.py script be able to figure out the default?

Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py:19]: Or teaching the CI machines to figure this out and provide the answer when they call the script? Maybe that's the best strategy?

Comment (zanderso) [reply sky/tools/create_macos_gen_snapshots.py:19]: Yeah, we'll have to modify the recipe to get it to work on arm64.

Comment (flar) [reply sky/tools/create_macos_gen_snapshots.py:19]: Is it worth updating the recipes now?

Comment (zanderso) [reply sky/tools/create_macos_gen_snapshots.py:19]: After this lands, we should update to explicitly pass the new flag, but I think we should wait to add logic to programatically figure out the arch until we're ready to bring those machines online.

**Pull request merged**
